app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -227,41 +135,155 @@ private function parseReferralCode($referralCode)
                 'northstar_id' => null, // set the user to null so we force account creation when the code is not present.
                 'campaign_id' => 8017,
                 'campaign_run_id' => 8022,
-                'source' => 'rock-the-vote',
+                'source' => $this->source,
                 'source_details' => null,
             ];
         }
 
         return $finalValues;
     }
+    /**
+     * Translate a status from Rock The Vote into a status that can be sent to Rogue.
+     *
+     * @param  string $rtvStatus
+     * @param  string $rtvFinishWithState
+     * @return string
+     */
+    private function parseVoterRegistrationStatus($rtvStatus, $rtvFinishWithState)
+    {
+        $rtvStatus = strtolower($rtvStatus);
+        $rtvFinishWithState = strtolower($rtvFinishWithState);
+
+        if($rtvStatus === 'complete') {
+            if ($rtvFinishWithState === "no") {
+                return 'register-form';
+            }
+            if ($rtvFinishWithState === "yes") {
+                return 'register-OVR';
+            }
+        }
+
+        if (strpos($rtvStatus, 'step') !== false) {
+            return 'uncertain';
+        }
+
+        if ($rtvStatus === 'rejected' || $rtvStatus === 'under 18') {
+            return 'ineligible';
+        }
+
+        return '';
+    }
+
+}
+
+class CreateRockTheVotePostInRogue implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, ImportToRogue;
+
+    /**
+     * The record to be created into a post from the csv.
+     *
+     * @var array
+     */
+    protected $data;
+    protected $record;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($record)
+    {
+        $this->data = new RockTheVoteRecord($record);
+        // TODO: Remove once all $this->record references have been replaced with $this->data.
+        $this->record = $record;
+    }
 
     /**
-     * If userId param is given, check for valid user.
-     * Otherwise check by job record email, mobile.
+     * Execute the job to create a Turbo Vote post in Rogue.
+     *
+     * @return array
+     */
+    public function handle(Rogue $rogue)
+    {
+        if (is_test_email($this->data->email)) {
+            info('progress_log: Skipping test: ' . $this->data->email);
```

Comment:
@ngjo - This PR keeps the logic for checking for test emails, but removes the logic of looking for last names that include `Baloney`. Do we need to keep the check for test last name?

Reply:
@aaronschachter yea, I think we can remove that check for `Baloney` now

Reply:
Oh jeeeeez haha, yeah glad we're removing that. 😂